crates/backend/src/main.rs

```diff
@@ -38,9 +42,84 @@ async fn main() -> std::io::Result<()> {
                     .header("Cross-Origin-Opener-Policy", "same-origin")
                     .header("Cross-Origin-Embedder-Policy", "require-corp"),
             )
-            .service(fs::Files::new("/", &frontend_folder).index_file("index.html"))
+            .service(serve_frontend(&frontend_folder))
     })
     .bind(format!("127.0.0.1:{}", port))?
     .run()
     .await
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use actix_web::{
+        http,
+        test,
+        App,
```

Comment:
Just a minor, but we don't need to import App for the tests, its already included by `use super::*;`